Userland/Libraries/LibCore/System/Promises.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+#include <AK/Vector.h>
+#include <Kernel/API/Pledge.h>
+#include <initializer_list>
+
+namespace Core::System {
+
+struct Promises {
+
+#define __ENUMERATE_PLEDGE_PROMISE(promise_name) bool promise_name : 1 { false };
+    ENUMERATE_PLEDGE_PROMISES
+#undef __ENUMERATE_PLEDGE_PROMISE
+
+    String to_string() const
```

Comment:
This is a very dissatisfying replacement imo, just C++-ifying something should not require going from no allocations to a whole bunch of allocations.

Perhaps consider
```
Pledge<Pledge::foo, Pledge::bar, Pledge::baz>();
```
and generate the string at compiletime? I'm sure you can make it work with a bit of tinkering.
?